jupyter_client/ioloop/manager.py

```diff
@@ -15,7 +16,17 @@
 def as_zmqstream(f):
     def wrapped(self, *args, **kwargs):
         socket = f(self, *args, **kwargs)
-        return ZMQStream(socket, self.loop)
+        save_socket_type = None
+        # zmqstreams only support sync sockets
+        if self.context._socket_type is not zmq.Socket:
+            save_socket_type = self.context._socket_type
+            self.context._socket_type = zmq.Socket
+        try:
+            return ZMQStream(socket, self.loop)
+        finally:
+            if save_socket_type:
+                # restore default socket type
+                self.context._socket_type = save_socket_type
```

Comment:
Sorry! This is what I get for having too much mypy on the brain. It's `socket_class` not `socket_type`

```suggestion
        save_socket_class = None
        # zmqstreams only support sync sockets
        if self.context._socket_class is not zmq.Socket:
            save_socket_class = self.context._socket_class
            self.context._socket_class = zmq.Socket
        try:
            return ZMQStream(socket, self.loop)
        finally:
            if save_socket_class:
                # restore default socket class
                self.context._socket_class = save_socket_class
```